indico/modules/rb_new/controllers/backend/bookings.py

```diff
@@ -260,8 +260,13 @@ def _process(self, args):
             'repeat_frequency': args['repeat_frequency'],
             'repeat_interval': args['repeat_interval'],
         }
-
+        has_date_changed = not self.booking.is_equal_with(data)
         self.booking.modify(data, session.user)
+
+        room = self.booking.room
+        if (has_date_changed and room.reservations_need_confirmation and self.booking.is_accepted and
```

Comment:
instead of checking if the room generally uses prebookings i'd check `not can_book and can_prebook`